SofaKernel/framework/sofa/defaulttype/Mat.h

```diff
@@ -342,10 +360,10 @@ class Mat : public helper::fixed_array<VecNoInit<C,real>,L>
         return m;
     }
 
-    /// Transpose current matrix.
+    /// Transpose the square matrix.
     void transpose()
     {
-        static_assert(L == C, "");
+        assert(L == C && "Cannot self-transpose a non-square matrix. Use transposed() instead");
```

Comment:
If i'm not mistaken the static_assert will be triggered for any non-square matrix regardless of transpose being called or not.

It is possible to make this check statically but it requires more boilerplate.